up-l1/someip.adoc

```diff
@@ -0,0 +1,221 @@
+
+.Sink & Source URI Mapping
+[sink-source-uri-mapping]
+|===
+|*UUri Portion* |*SOME/IP Field* |*Description*
+|*`UAuthority::ip`* |IP address & port |IP address & port (destination
+endpoint) of the mDevice
+
+|*`UEntity::id`* |Service ID |Software entity ID that is placed in the upper 16
+bits of the SOME/IP Message ID header field
+
+|*`UResource::id`* |Method/Event ID |Identifier of the method/resource placed in the
+lower 15 bits of the SOME/IP Message ID header field
+
+|===
+
+
+==== UUIDs
+
+link:../basics/uuid.adoc[uProtocol UUID specifications] create a unique identifier for each message along with timestamp information. the ID is used for correlate between request and response as well. SOME/IP instead defines the ID as 16bit client ID + 16 bit session ID (that is incremented). When messages are converted SOME/IP to/from uProtocol, care must be taken to ensure that the SOME/IP ID and uProtocol UUID are properly mapped, especially when corelating a request to a response. 
+
+* Generated SOME/IP events *SHOULD* set the 16 bit client ID to 0 per SOME/IP specifications 4.1.2.4
+* *MUST* auto-populate the request ID cached from the request event, into the response event and then flush the entry in the cache
+
+Further details of the usage of IDs for the various message types can be found in the next section.
+
+==== Message Type
+
+.UMessageType Mapping
+[umessagetype-mapping]
+|===
+| UMessageType | SOME/IP Type | Details
+
+|`UMESSAGE_TYPE_PUBLISH` or `UMESSAGE_TYPE_NOTIFICATION` |`NOTIFICATION` |Both publication and notifications events are the same for SOME/IP
+
+| `UMESSAGE_TYPE_REQUEST` |`REQUEST` |Requests type message
+
+| `UMESSAGE_TYPE_RESPONSE` |`RESPONSE` or `ERROR` |Responses or error has occurred while attempting to deliver the message or a service
+has thrown an exception
+
+|===
+
+
+When receiving uProtocol initiated requests:
+
+* *MUST* cache the request `UAttributes` for a maximum of `ttl` so that it can be used to build a response `UAttributes` when receiving a response from SOME/IP
+
+```
+response.priority = request.priority
+response.reqid = request.id
+```
+
+* Auto-generated SOME/IP REQUEST message, *MAY* set the 16 bit client ID to the 0 per SOME/IP specifications 
+
+When receiving a SOME/IP initiated requests:
+
+* *MUST* cache the SOME/IP request ID as well as the generated `UAttributes` for the request messages so that the response can be translated back to a SOME/IP response message
+
+
+==== Communication Status 
+
+<<commstatus-mapping>> below provides the mapping of uAttributes `commstatus` `UCode` codes to SOME/IP error codes.
+
+.Error Code Mapping
+[commstatus-mapping, width=70%]
+|===
+| UCode | SOME/IP Error Codes
+
+|`OK` |`E_OK`
+|`INVALID_ARGUMENT` |`E_WRONG_MESSAGE_TYPE` or `E_UNKNOWN_METHOD`
+|`DEADLINE_EXCEEDED` |`E_TIMEOUT`
+|`NOT_FOUND` |`E_UNKNOWN_SERVICE`
+|`UNAVAILABLE` |`E_NOT_READY`
+|`DATA_LOSS` |`E_MALFORMED_MESSAGE`
+|`INTERNAL` |`E_NOT_REACHABLE`
```

Comment:
Also add: `E_NOT_OK` here

- `E_WRONG_PROTOCOL_VERSION`, `E_WRONG_INTERFACE_VERSION` not described as well

For more error codes add link to SOME/IP Protocol specification: `[PRS_SOMEIP_00191]`

Reply:
done, @int0x27 please check